openedx/openedx-platform
# feat:  keep learner state associated after libraries migration

## Description
Upgrading an LCB's source lib from V1 to V2 disassociates learner state, because it seems child use keys are derived differently between V1 and V2.

## Supporting information
https://twou.slack.com/archives/C05NT2YN820/p1702047279641349

## Testing instructions
1. Create an LCB
2. Create a v1 library
3. Associate the V1 library with the LCB. set max_count to 1.
4. View the LCB in LMS (generating learner state for you). note the LC block's children for later comparison using "debug info".
6. Migrate the v1 library to v2 using `./manage.py cms copy_libraries_from_v1_to_v2 '11111111-2111-4111-8111-111111111111' ./list_of--library-locators.csv <v1 lib id>`
7. Switch the LCB to be pointing at the created v2 lib by changing the source lib attribute.
8. View the LCB in LMS, you should see the same 1 block from before and the children should be identical, but the source_library_id is a v2 lib
